### PR TITLE
restart the application manager controller if there is a hub kubeconf…

### PR DESCRIPTION
**Pull request by xiangjingli**
…ig change

Signed-off-by: Xiangjing Li <xiangli@redhat.com>

https://github.com/stolostron/backlog/issues/26144#issuecomment-1271924852

### CODE REVIEW

File: cmd/manager/exec/manager.go [hunk 1]
```diff
@@ -213,16 +211,42 @@ func RunManager() {
...
 
-		go wait.JitterUntilWithContext(context.TODO(), leaseReconciler.Reconcile,
-			time.Duration(Options.LeaseDurationSeconds)*time.Second, leaseUpdateJitterFactor, true)
+		ctx, cancel := context.WithCancel(context.TODO())
+
+		go func() {
+			for {
+				err := leaseReconciler.CheckHubKubeConfig(ctx)
+				if err != nil {
+					cancel()
```
Comment (mikeshng) [inline cmd/manager/exec/manager.go:236]: What happens if this line does execute? Will it kill this goroutine and we lose this lease reconcile?

**Review by mikeshng — COMMENTED** [review]
The changes look good to me just one minor concern.

@philipwu08 I think you wrote similar code recently in hypershift addon, if you have some cycle could you please take a quick look at this change? Thanks.

File: cmd/manager/exec/manager.go (hunk 1 above)
Comment (xiangjingli) [reply cmd/manager/exec/manager.go:236]: Not only the goroutine, bu the whole controller will exit.  

the cancel() is called when there is kube kubeconfig change. Then it stops the context channel `ctx.Done()`.  As a result,  os.Exit is invoked to exit the whole controller for restart.

**Comment by xiangjingli** [comment]
@philipwu08  As Mike commented, could you help review the PR as there are some similar code change in hypershift addon.

**Pull request merged**